### PR TITLE
fix: the 'belongedClass' property of the overridden method has been tampered with

### PR DESCRIPTION
**Pull request by Shanks0224**
When the method of the base class is **overridden**, the 'belongingClass' property of this method is tampered with into its subclass.

### CODE REVIEW

**Comment by Shanks0224** [comment]
Currently, the ‘belongedClass‘ is used in the specialization logic of the constructor of a generic class.
Since there is no override relationship between the constructor of the subclass and the constructor of the base class, the ‘belongedClass‘ of the base class constructor will not be tampered with.

File: src/type.ts [hunk 1]
```diff
@@ -480,7 +480,9 @@ export class TSClass extends TSTypeWithArguments {
 
     addMethod(classMethod: TsClassFunc): void {
         classMethod.type.isMethod = true;
-        classMethod.type.belongedClass = this;
+        // the subclass inherits the method of the base class and should not modify the belongedClass of the base class method.
```
Comment (zhenweijin) [inline src/type.ts:483]:
```suggestion
        // when sub class inherits the method of the base class,  it should not modify the `belongedClass`.
```

File: src/type.ts [hunk 2]
```diff
@@ -1893,7 +1910,11 @@ export class TypeResolver {
                 infc.addMemberField(field);
             }
             for (const method of baseInfcType.memberFuncs) {
-                infc.addMethod(method);
+                infc.addMethod({
+                    name: method.name,
+                    type: method.type.clone(),
```
Comment (zhenweijin) [inline src/type.ts:1915]: why here use a copy type of method?

File: src/type.ts (hunk 1 above)
Comment (Shanks0224) [reply src/type.ts]: done

**Review by zhenweijin — COMMENTED** [review]
LGTM

**Pull request merged**